providers.tf

```diff
@@ -0,0 +1,22 @@
+terraform {
+  required_version = ">= 0.15.0"
+  experiments      = [module_variable_optional_attrs]
```

Comment:
Mildly concerned with using experimental features in a project with big potential like this one, but not a blocker.